src/redux/books/booksSlice.js

```diff
@@ -18,22 +20,27 @@ const initialState = [
     title: 'The Selfish Gene',
     author: 'Richard Dawkins',
     category: 'Nonfiction',
-  },
-];
+  },];
+
+export const fetchBooks = createAsyncThunk('books/fetchBooks', async () => {
+  const response = await axios.get('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/abc123/books');
+  return response.data;
+});
+
```

Comment:
- [ ] Hi @Bupilipili, now we should connect this external API to our existing React-Redux application by refactoring codes in such a way that adding and removing books is done on a remote server(API). You did a nice job creating  the fetchBooks action creator but this should exist in the bookSlice. fetchBooks action creator should actually updates the state of bookSlice and make sure useEffect is use when dispatching it in the BookList component. addBook and removeBook should be refactor to return a promise, it should add, remove new book through the API and authomatically dispatch an action type that will add, remove a book to the books list and updates the state. Please kindly fix :wrench:  the issue.

Reply:
Thanks for the insight 💯 am on it 😃